src/wazuh_modules/wm_exec.c

```diff
@@ -690,22 +680,22 @@ void wm_kill_children() {
 
                     // If time is gone, kill process
 
-                    mdebug1("Killing process group %d", (int)sid);
+                    mdebug1("Killing process group %d", sid);
 
                     kill(-sid, SIGKILL);
                     exit(EXIT_SUCCESS);
 
                 default: // Parent
-                    wm_children[i] = 0;
+                    break;
+
                 }
             } else {
                 // Kill immediately
                 kill(-sid, SIGKILL);
             }
         }
     }
-
-    w_mutex_unlock(&wm_children_mutex);
+    wm_children_pool_destroy();
```

Comment:
do we need this at all if we can not use the same over again? adding extra checks for this is a good idea if we need to.